### PR TITLE
fix(knex): nested queries result in wrong span names

### LINKED ISSUE #1535: instrumentation-knex table calculated incorrectly on nested tables
### What version of OpenTelemetry are you using?

@opentelemetry/instrumentation: 0.39.1

### What version of Node are you using?

16.14.0

### What did you do?
Used opentelemetry-knex and with nested queries, it misbehaves, potentially generating a long span name.

See https://github.com/nachogiljaldo/opentelemetry-js-contrib/commit/95219640a7c8bdfb189c5b094db44eb645282cb0 for reproduction steps.

### What did you expect to see?
The code should work somewhat recursively to find the first table. And return a table (and not a subquery as was the case).

### What did you see instead?
the variable table was set to be a queryBuilder, when serialized that resulted on the whole query

### Additional context
The linked commit adds a potential solution, if the solution looks good I'm happy to create a PR out of it.

### PR DESCRIPTION
**Pull request by nachogiljaldo**
Fixes #1535

## Which problem is this PR solving?

- instrumentation-knex generates wrong span names when the knex query has a nested query in its first element

## Short description of the changes

- consider the possibility of nested queries and add the corresponding unit tests to deal with it

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-knex/src/utils.ts [hunk 1]
```diff
@@ -80,3 +80,11 @@ export const limitLength = (str: string, maxLength: number) => {
   }
   return str;
 };
+
+export const extractTableName = (builder: any): string => {
```
[the remark this answers is not in the document]
Comment (nachogiljaldo) [reply plugins/node/opentelemetry-instrumentation-knex/src/utils.ts:84]: I had a look at knex and I couldn't find any exported type that I could use having _single. I could define my own custom internal type, but I'd rather not do it, WDYT?

**Comment by nachogiljaldo** [comment]
Hey @rauno56 I believe your review is requested in order to merge this. Any chance you can have a look in the near future?

File: plugins/node/opentelemetry-instrumentation-knex/src/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pichlermarc) [reply plugins/node/opentelemetry-instrumentation-knex/src/utils.ts:84]: > I had a look at knex and I couldn't find any exported type that I could use having _single. I could define my own custom internal type, but I'd rather not do it, WDYT?

Makes sense, I think as we deal with any in `wrapped_logging_method` already we can also deal with using any here. :+1:

[the remark this answers is not in the document]
Comment (nachogiljaldo) [reply plugins/node/opentelemetry-instrumentation-knex/src/utils.ts:84]: are we good to resolve and merge then?

**Comment by nachogiljaldo** [comment]
@rauno56 any chance you have a look at this?

File: plugins/node/opentelemetry-instrumentation-knex/test/index.test.ts [hunk 2]
```diff
@@ -190,6 +190,275 @@ describe('Knex instrumentation', () => {
...
+                parentSpan,
+              },
+            ]);
+            assert.strictEqual(instrumentationSpans[0].name, ':memory:');
+            assert.strictEqual(
+              instrumentationSpans[1].name,
+              'insert :memory:.testTable1'
+            );
+
+            assert(last.name, 'parentSpan');
```
Comment (blumamir) [inline plugins/node/opentelemetry-instrumentation-knex/test/index.test.ts:457]: What are these asserts meant to verify in this context?

**Review by blumamir — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-knex/test/index.test.ts (hunk 2 above)
Comment (nachogiljaldo) [reply plugins/node/opentelemetry-instrumentation-knex/test/index.test.ts]: I removed them, the first block is somewhat redundant and the second one, I personally believe it was not really relevant.

**Review by blumamir — APPROVED** [review]

**Comment by nachogiljaldo** [comment]
@blumamir I am afraid I can't merge because @rauno56 is still requested and I can't remove him. Anything you can do?

**Comment by blumamir** [comment]
> @blumamir I am afraid I can't merge because @rauno56 is still requested and I can't remove him. Anything you can do?

It's ok I merged it 👍

**Pull request merged**